### DISCUSSION TITLE
Gamification

### DISCUSSION THREAD

**Discussion by raphodn** [openfoodfacts/open-prices#399] [General]
### Existing

- list of top contributors : https://prices.openfoodfacts.org/users

#### App wide (home page ?)

Some stats for the ongoing week (week number ?)
- top 3 users (podium)
- number of new prices
- number of new products
- number of new locations

Stats app + AllTimeStats model + MonthlyStats model + WeeklyStats model

#### Per user

|Category|Description||
|---|---|---|
|Prices|Top 10 contributor badge||
|Product|Unlock: user was the first to add a price|new `price_first_added_by`  field|
|Product|Mayor: user has added the most prices for this product||
|Brand|Mayor: user has added the most prices for this brand||
|Location|Unlock (price tag proof): user was the first to upload a PRICE_TAG proof|new `proof_price_tag_first_added_by` field|
|Location|Unlock (receipt proof): user was the first to upload a RECEIPT proof|new `proof_receipt_first_added_by` field|
|Location|Unlock: user was the first to add a price|new ` price_first_added_by`  field|
|Location|Mayor: user has added the most prices for this location|new `price_top_added_by` field|

Show a popup if new location or product ?

**Comment by serpico** [comment]
about stats, resources and maybe gamification https://github.com/openfoodfacts/smooth-app/issues/6031#issuecomment-2531641730

**Comment by raphodn** [comment]
Notification ?

**Comment by raphodn** [comment]
Badges (1, 10, 100) vs leaderboard

**Comment by raphodn** [comment]
Strikes

**Comment by raphodn** [comment]
We're running a super quick survey (1mn) to get some ideas from the community, regarding gamification in Open Prices.

:point_right: https://framaforms.org/gamification-in-open-prices-1780570507

Thanks!

**Reply by raphodn** [reply]
preliminary results 2 weeks later

<img width="770" height="831" alt="image" src="https://github.com/user-attachments/assets/e87b56c3-c887-48eb-9358-d6d65dcc4fbb" />

top 2 ideas
- end of the year recap: we have some time to work on this
- badges!

**Comment by raphodn** [comment]
## Badges

What kind of badges would you like to see?
Do you have examples of apps or (community contribution) websites that do things neatly?

## What is currently implemented

_last updated: 2026-08-21_

|name|field|steps|
|-|-|-|
|Contributor|price_count|1 / 10 / 100 / 1000 / 10000|
|Shop owner|proof_type_shop_import_count|1|
|Traveller|location_count|2 / 10 / 50 / 200|
|Globetrotter|location_type_osm_country_count|2 / 5 / 10|
|Challenger|challenge_count|1 / 5 / 10|
|Historian|year_count|2 / 5 / 10|

19 badges in total

**Reply by raphodn** [reply]
cc @Freso you mentioned a digital common doing this the other day?

Discourse has a badge system: https://forum.openfoodfacts.org/badges
- grouped by "Getting started", "Community", "Posting", "Trust level"
- seems to be public (in terms of listing + users awarded)

<img width="1087" height="428" alt="image" src="https://github.com/user-attachments/assets/a38762dc-e4f3-4944-99aa-67b72b731fdb" />

**Reply by Freso** [reply]
That was for "year in review", and I linked it in the meeting notes :)

For badges, BookBrainz does badges: https://bookbrainz.org/ (e.g., [my badges](https://bookbrainz.org/editor/23/achievements))

**Reply by raphodn** [reply]
nice !

<img width="1080" height="593" alt="image" src="https://github.com/user-attachments/assets/4d3cb3d6-cf88-479c-827c-9be73f4fa23a" />

I wonder if they have also a page that lists all the available badges + number of users who achieved them

also : https://www.youtube.com/watch?v=XEZROTq1lY0 (they built in 2016 with GSoC)

**Reply by TTalex** [reply]
Some ideas, extracted from a private Slack chat:

- Buddy : Added X prices from someone else's proofs
- Quality patrol : Flag a wrong price or proof
- Swiss Army person : Contribute using X different tools (web, mobile, api, rgpd)
- Certified reuser : Have a project listed in the known reuses
- Early bird : Be the first to add a proof on a challenge
- Beta-tester : Use a feature while it's still an experiment
- Product creator : Add X missing products using Open Prices
- Hard worker : Add X prices in a single day (minimum 100 or 200)
- Contributing month-ster : Be the most active contributor in a given month
- On a roll : Contribute for X consecutive months
- Impact seeker : X Prices added to most scanned products
- Completist : Add X prices in each Open Food Facts flavors
- Camper : Add X prices in a single location
- Regular : Take pictures X different days in the same location
- Sponsor : Donate to support Open Food Facts (not sure how to track this one, and if we want to, since it's a bit pay to win)
- Hacker : Contribute on GitHub (hard to track automatically, Translator would be the same with crowdin)

**Reply by Freso** [reply]
> Early bird : Be the first to add a proof on a challenge

Maybe “be among the first” and count the first 24 hours of a challenge? Or the first 1% of one (in case we want to do "spot"/"lightning"/short-lived challenges (like for a day or a few days)) maybe.

**Reply by raphodn** [reply]
easy-to-do badges with the current (and recent) user counts
- Buddy (or Validator ? requires explanation in any case :) - using price_in_proof_not_owned_count
- Tourist - using `location_type_osm_city_count` (duplicate with Traveller ?)
- Globetrotter - using `location_type_osm_country_count`
- Shopper - using `proof_type_receipt_count`
- ??? - using `proof_type_gdpr_request_count`
- Shop owner - using `proof_type_shop_import_count`

**Reply by raphodn** [reply]
and probably start displaying them in "categories" like discourse :eyes:
